packages/core/contracts/oracle/implementation/OptimisticOracle.sol

```diff
@@ -597,11 +644,15 @@ contract OptimisticOracle is OptimisticOracleInterface, Testable, Lockable {
             payout
         );
 
+        // Temporarily disable the re-entrancy guard early to allow the caller to take an OO-related action inside this callback.
+        _notEntered = true;
         // Callback.
         if (address(requester).isContract())
             try
                 OptimisticRequester(requester).priceSettled(identifier, timestamp, ancillaryData, request.resolvedPrice)
             {} catch {}
+        // Re-enable this guard since the calling function may take other actions that need to be guarded after this.
+        _notEntered = false;
```

Comment:
Why not set this to false in the other locations where you set `_notEntered = true` before a callback?

Reply:
This method is called inside of another method, so there's other code that can run after this.

In the other cases, it's immediately followed by the end of the function. I can set it in those cases too just to be safe and consistent, though.

Reply:
Even though it's a little redundant, I tried to make this a little clearer and consistent. Let me know what you think.

Reply:
think better to be a bit consistent

Reply:
actually maybe its just worth saving the gas since its clear its near end of function. Instead of closing it you could replace it with a comment explaining why you don't reset _notEntered. WDYT?